test/unit/models/consent.test.ts

```diff
@@ -0,0 +1,207 @@
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Knex from 'knex'
+import Config from '../../../config/knexfile'
+import ConsentDB, { Consent } from '../../../src/model/consent'
+
+/*
+ * Mock Consent Resources
+ */
+const partialConsent: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123'
+}
+
+const completeConsent: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123',
+  credentialId: 123,
+  credentialType: 'FIDO',
+  credentialStatus: 'PENDING',
+  credentialChallenge: 'xyhdushsoa82w92mzs',
+  credentialPayload: 'dwuduwd&e2idjoj0w'
+}
+
+/*
+ * Consent Resource Model Unit Tests
+ */
+describe('src/model/consent', (): void => {
+  let Db: Knex
+  let consentDB: ConsentDB
+
+  beforeAll(async (): Promise<void> => {
+    Db = Knex(Config.test)
+    await Db.migrate.latest()
+
+    consentDB = new ConsentDB(Db)
+  })
+
+  afterAll(async (): Promise<void> => {
+    Db.destroy()
+  })
+
+  describe('register', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+    })
+
+    it('adds consent with partial info to the database', async (): Promise<void> => {
+      // Action
+      await consentDB.register(partialConsent)
+
+      // Assertion
+      const consents: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: partialConsent.id
+        })
+
+      delete consents[0].timeStamp
+
+      expect(consents[0]).toEqual({
+        id: '1234',
+        initiatorId: 'pisp-2342-2233',
+        participantId: 'dfsp-3333-2123',
+        credentialId: null,
+        credentialType: null,
+        credentialStatus: null,
+        credentialPayload: null,
+        credentialChallenge: null
+      })
+    })
+
+    it('returns an error on adding a consent with existing consentId', async (): Promise<void> => {
```

Comment:
No need to have `Promise<void>` here... it's implied

Reply:
for me linter is complaining, maybe let update the rule?

Reply:
But we are using `Promise<type>` as the return type for all async functions. Why not here?

Reply:
Yeah the linter wants return types for anonymous as well as named functions. Isn't that the convention though?

Reply:
@lewisdaly would you agree to keep strictly typed return types? It looks little strange for some obvious cases, but we already have it used in so many places ;-)

Reply:
Ok no worries, yeah I just thought it looked strange, no major qualms here.